config/initializers/devise.rb

```diff
@@ -260,6 +260,8 @@
   # Add a new OmniAuth provider. Check the wiki for more information on setting
   # up on your models and hooks.
   # config.omniauth :github, 'APP_ID', 'APP_SECRET', scope: 'user,public_repo'
+  config.omniauth :facebook, "2628485857246281", "d9fa8569e679d978846467a72f98e663", callback_url: "http://localhost:3000/users/auth/facebook/callback"
```

Comment:
There's a more safer way to do this by hiding your app key and secret. You can check out the following gems: Figaro and Dotenv. I used the former because it was built with rails and deploying to heroku in mind.

This section of your code would in turn look like this (take a look at the scope, which means it is only fetching email and birthday):

```
config.omniauth :facebook, ENV['facebook_app_id'], ENV['facebook_app_secret'], scope: "email, user_birthday"
```

And over in a new file in `config/initializers/figaro.rb`:

```
Figaro.require_keys("facebook_app_id", "facebook_app_secret")
```

And of course you'd save your credentials in a file that won't be commited to the DDBB.

Another tool to consider is Rails Credentials which come built in with Rails, as of 5.2, and Rails 6 makes it easier to configure.

Reply:
By the way, this one is completely OPTIONAL and NOT REQUIRED